pkg/catalog/catalog.go

```diff
@@ -1000,6 +1000,40 @@ func (c *Catalog) GetEntry(ctx context.Context, repositoryID string, reference s
 	return &catalogEntry, nil
 }
 
+// UpdateEntryUserMetadata updates user metadata for the current entry for a
+// path in repository branch reference.
+func (c *Catalog) UpdateEntryUserMetadata(ctx context.Context, repositoryID, branch, path string, newUserMetadata map[string]string) error {
+	branchID := graveler.BranchID(branch)
+	if err := validator.Validate([]validator.ValidateArg{
+		{Name: "repository", Value: repositoryID, Fn: graveler.ValidateRepositoryID},
+		{Name: "branch", Value: branchID, Fn: graveler.ValidateBranchID},
+		{Name: "path", Value: Path(path), Fn: ValidatePath},
+	}); err != nil {
+		return fmt.Errorf("%w: %s %s %s", err, repositoryID, branchID, path)
+	}
+
+	repository, err := c.getRepository(ctx, repositoryID)
+	if err != nil {
+		return nil
+	}
+
+	key := graveler.Key(path)
+	updater := graveler.ValueUpdateFunc(func(value *graveler.Value) (*graveler.Value, error) {
```

Comment:
I think I have forgotten to submit this comment in the previous review.
It also relates to @itaiad200  [comment](https://github.com/treeverse/lakeFS/pull/8264/files#r1793663660):
If the intention of the function is to update the metadata only - let's pass the metadata instead of an update function. make it explicit, inline and easier to read.
Otherwise lets modify the graveler function to be `UpdateObject` and provide the update function accordingly (which is infact exactly the same as calling `g.StagingManager.Update`).